### ISSUE TITLE
Seperate out general poolParams in gamm module

### ISSUE THREAD

**Issue by mattverse** [osmosis-labs/osmosis#683]
Currently gamm module is designed in the way that every pool-model being implemented would have their own params struct which would contain exitFee and swapFee.
There's a room to further generalize gamm module by creating a generalized poolParams struct that would contain swapFee and exitFee that all pool models could implement.

**Comment by mconcat** [comment]
I agree with this. Common params scattered around different pool models will make it hard to manage for example swapfee by the governance.
